hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/keyvalue/KvMetadataResolvers.java

```diff
@@ -116,25 +115,25 @@ public List<MappingField> resolveAndValidateFields(
             }
         }
 
+        final String keyFormat = getFormat(options, true);
+        if (NESTED_FIELDS_SUPPORTED_FORMATS.contains(keyFormat)) {
+            extractFields(userFields, true).values().forEach(mappingField ->
+                    enrichMappingFieldType(true, mappingField, ss, relationsStorage, options));
```

Comment:
`enrichMappingFieldType` checks key/value format for each field. It works, but it might be cleaner to create appropriate `FieldEnricher` once and then pass all key or value fields (respectively) to single enricher.